run/gem5art/run.py

```diff
@@ -463,18 +513,22 @@ def handler(signum, frame):
 
             if self.current_time - self.start_time > self.timeout:
                 proc.kill()
-                self.kill_reason = 'timeout'
+                self.kill_reason = "timeout"
 
             if self.checkKernelPanic():
                 proc.kill()
-                self.kill_reason = 'kernel panic'
+                self.kill_reason = "kernel panic"
 
-            self.dumpJson('info.json')
+            if self.check_failure(self):
```

Comment:
Can you add a comment here explaining that since this is a user-provided function we have to pass self in (it's not bound).